Python/symtable.c

```diff
@@ -2798,6 +2822,16 @@ symtable_handle_comprehension(struct symtable *st, expr_ty e,
     if (!symtable_exit_block(st)) {
         return 0;
     }
+    if (is_async &&
+        !(st->st_cur->ste_type == FunctionBlock && st->st_cur->ste_coroutine) &&
```

Comment:
Perhaps

`#define IS_ASYNC_DEF (st->st_cur->ste_type == FunctionBlock && st->st_cur->ste_coroutine)`

and use it here and in lines 2295-6?